src/Commands/Files/ConvertFileToPdf.cs

```diff
@@ -0,0 +1,181 @@
+        public SwitchParameter Force;
+
+        [Parameter(Mandatory = true, ParameterSetName = UPLOADTOSHAREPOINT)]
+        [ValidateNotNullOrEmpty]
+        public FolderPipeBind DocumentLibrary;
+
+        [Parameter(Mandatory = false, ParameterSetName = UPLOADTOSHAREPOINT)]
+        public SwitchParameter Checkout;
+
+        [Parameter(Mandatory = false, ParameterSetName = UPLOADTOSHAREPOINT)]
+        public string CheckInComment = string.Empty;
+
+        [Parameter(Mandatory = false, ParameterSetName = UPLOADTOSHAREPOINT)]
+        public CheckinType CheckinType = CheckinType.MinorCheckIn;
+
+        [Parameter(Mandatory = false, ParameterSetName = UPLOADTOSHAREPOINT)]
+        public SwitchParameter Approve;
+
+        [Parameter(Mandatory = false, ParameterSetName = UPLOADTOSHAREPOINT)]
+        public string ApproveComment = string.Empty;
+
+        [Parameter(Mandatory = false, ParameterSetName = UPLOADTOSHAREPOINT)]
+        public SwitchParameter Publish;
+
+        [Parameter(Mandatory = false)]
+        public string PublishComment = string.Empty;
+
+        [Parameter(Mandatory = false, ParameterSetName = URLASMEMORYSTREAM)]
+        public SwitchParameter AsMemoryStream;
+
+        protected override void ExecuteCmdlet()
+        {
+            if (string.IsNullOrEmpty(Path))
+            {
+                Path = SessionState.Path.CurrentFileSystemLocation.Path;
+            }
+            else if (!System.IO.Path.IsPathRooted(Path))
+            {
+                Path = System.IO.Path.Combine(SessionState.Path.CurrentFileSystemLocation.Path, Path);
+            }
+
+            var graphAccessToken = TokenHandler.GetAccessToken(this, $"https://{Connection.GraphEndPoint}/.default", Connection);
+            Url = Utilities.UrlUtilities.UrlDecode(Url.Replace("+", "%2B"));
+            var serverRelativeUrl = GetServerRelativeUrl(Url);
+            var fileObj = GetFileByServerRelativePath(serverRelativeUrl);
+            var siteId = PnPContext.Site.Id.ToString();
+            var listId = fileObj.ListId.ToString();
+            var itemId = fileObj.ListItemAllFields.Id.ToString();
+            var sourceFileName = fileObj.Name.ToString().Substring(0, fileObj.Name.ToString().LastIndexOf("."));
+            var apiUrl = $"https://{Connection.GraphEndPoint}/v1.0/sites/{siteId}/lists/{listId}/items/{itemId}/driveItem/content?format=pdf";
+            byte[] response = RestHelper.GetByteArrayAsync(Connection.HttpClient, apiUrl, graphAccessToken).GetAwaiter().GetResult();
+            var fileToDownloadName = !string.IsNullOrEmpty(sourceFileName) ? sourceFileName : "Download";
+
+            switch (ParameterSetName)
+            {
+                case URLTOPATH:
+                    var fileOut = System.IO.Path.Combine(Path, $"{fileToDownloadName}.pdf");
+                    if (!Directory.Exists(Path))
+                    {
+                        Directory.CreateDirectory(Path);
```

Comment:
It should throw error if Folder doesn't exist.

Reply:
Changes done in the latest commit